src/main/msp/msp.c

```diff
@@ -2466,6 +2475,21 @@ static mspResult_e mspProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
 #endif
         }
 
+        if (sbufBytesRemaining(src) >= 1) {
+            // Added in MSP API 1.42
+            // Read upper byte of the cutoff value. This version of MSP's increased the maximum value from 8 bit to 16
+#ifdef USE_RC_SMOOTHING_FILTER
+            rcSmoothingDerivativeCutoff |= (sbufReadU8(src) << 8);
```

Comment:
Not sure if this is the correct way to go. I think is better to let the code for the version 1.41 as is, for client with old versions, and read it complete (not only the upper byte) for the new version and overwrite the old value. It will be easy to be used for any client, not read half of the value in each place.

Reply:
`configRebootUpdateCheckU16` sounds like it'll flag for a reboot if something is changed, so it makes sense to move the call to it further down. But I agree that splitting the value is bad for readability purposes, the whole value should be sent. Like `rcSmoothingDerivativeCutoff = sbufReadU16(src);`

Reply:
Yes this is really ugly (see Open Questions section), although this was the only way to make it backwards compatible with existing configurator. If we go with `sbufReadU16` you can't use existing configurator and it'll mess up the configs as soon as you connect to it. Is there any way to prevent that?

Reply:
The changes to MSP shouldn't be included anyway. Our policy is to not update MSP or the Configurator until one release after a feature is added.

Reply:
I think we will need to include them for this, as it happened with other features. If not, when the user saves something in the receiver page it will overwrite this value without being aware of it.
When a user upgrades to a new firmware he must upgrade the Configurator too, so it is not a problem if you simply replace the value.

Reply:
I agree, this change has to be made. And since it's an extension of an existing parameter (that should be in a separate pull request though), that is fine.
But you should read the whole `uint16_t` here, and, if it is sent by the client, replace the `uint8_t` that was sent earlier.